python/amici/petab_import_pysb.py

```diff
@@ -0,0 +1,333 @@
+"""
+PySB-PEtab Import
+------------
+Import a model in the PySB-adapted :mod:`petab`
+(https://github.com/PEtab-dev/PEtab) format into AMICI.
+"""
+
+import logging
+import os
+from typing import List, Dict, Union, Optional, Tuple, Iterable
+
+import libsbml
+import petab
+import sympy as sp
+from petab.C import *
+
+from .logging import get_logger, log_execution_time, set_log_level
+from . import petab_import
+
+logger = get_logger(__name__, logging.WARNING)
+
+
+class PysbPetabProblem(petab.Problem):
+    """Representation of a PySB-model-based PEtab problem
+
+    This class extends :class:`petab.Problem` with a PySB model.
+    The model is augmented with the observation model based on the PEtab
+    observable table.
+    For now, a dummy SBML model is created which allows used the existing
+    SBML-PEtab API.
+
+    :ivar pysb_model:
+        PySB model instance from of this PEtab problem.
+
+    """
+
+    def __init__(self, pysb_model: 'pysb.Model' = None, *args, **kwargs):
+        """
+        Constructor
+
+        :param pysb_model: PySB model instance for this PEtab problem
+        :param args: See petab.Problem.__init__
+        :param kwargs: See petab.Problem.__init__
+        """
+
+        super().__init__(*args, **kwargs)
+
+        self.pysb_model: 'pysb.Model' = pysb_model
+        self._add_observation_model()
+
+        if self.pysb_model is not None:
+            self.sbml_document, self.sbml_model = \
+                create_dummy_sbml(
+                    self.pysb_model,
+                    observable_ids=self.observable_df.index.values
+                    if self.observable_df is not None else None
+                )
+
+    def _add_observation_model(self):
```

Comment:
Wouldn't this functionality ideally be provided by the PEtab library such that it could be reused by other tools?

Reply:
I find it hard to judge how reusable that is for other tools. In general, it could be added there, yes. Would maybe keep it here for now until this has been tested a bit more.